modules/git/repo_attribute.go

```diff
@@ -133,7 +135,14 @@ func (c *CheckAttributeReader) Init(ctx context.Context) error {
 		c.env = append(c.env, "GIT_WORK_TREE="+c.WorkTree)
 	}
 
-	c.env = append(c.env, "GIT_FLUSH=1")
+	if gitVersion.Equal(version.Must(version.NewVersion("2.43.1"))) {
+		// https://github.com/go-gitea/gitea/issues/29141  gitea hanging with git 2.43.1 #29141
+		// https://lore.kernel.org/git/CABn0oJvg3M_kBW-u=j3QhKnO=6QOzk-YFTgonYw_UvFS1NTX4g@mail.gmail.com/
+		// git 2.43.1 has a bug: the GIT_FLUSH polarity is flipped
+		c.env = append(c.env, "GIT_FLUSH=0")
+	} else {
+		c.env = append(c.env, "GIT_FLUSH=1")
+	}
```

Comment:
Why is this part not in the main branch PR?

Reply:
https://github.com/go-gitea/gitea/issues/29141#issuecomment-1939143849

I also think we do not need to do it in the main branch. 

Maybe it's good to help some unlucky users in 1.21 branch.